## nanoSPLITS
An image of every cell sorted by the cellenONE for nanoSPLITS analysis is included [here](https://github.com/Cajun-data/nanoSPLITS/tree/main/Cell_Sorting_Images). RNAseq metadata and a key for translating the various images and file names can be found [here](https://github.com/Cajun-data/nanoSPLITS/tree/main/Metadata). The RMarkdown files and scripts for each figure/experiment are provided. For the initial experiments involving C10 and SVEC cells, pre-rendered HTML files  can be accessed [here (pooled C10 cell experiment)](http://htmlpreview.github.io/?https://github.com/Cajun-data/nanoSPLITS/blob/main/Pooled_C10Cells/C10pooledcells_Figure2.html) and [here (SVEC and C10 single cells)](http://htmlpreview.github.io/?https://github.com/Cajun-data/nanoSPLITS/blob/main/SVEC_C10_Cells/C10SVEC_singlecells_Figure3.html).


# Parallel measurement of transcriptomes and proteomes from same single cells using nanodroplet splitting (nanoSPLITS)
## Authors
James M. Fulcher,1,* Lye Meng Markillie,1 Hugh D. Mitchell,1 Sarah M. Williams,1 Kristin M. Engbrecht,3 David J. Degnan,2 Lisa M. Bramer,2 Ronald J. Moore,2 William B. Chrisler,2 Joshua Cantlon-Bruce,4,5 Johannes W. Bagnoli,5 Wei-Jun Qian,2 Anjali Seth,5 Ljiljana Paša-Tolić,1 Ying Zhu1,6* <br>
1Environmental Molecular Sciences Laboratory, Pacific Northwest National Laboratory, Richland, Washington 99354, United States <br>
2Biological Sciences Division, Pacific Northwest National Laboratory, Richland, Washington 99354, United States <br>
3Nuclear, Chemistry, and Biology Division, Pacific Northwest National Laboratory, Richland, WA 99354, United States <br>
4Scienion AG, Volmerstraße 7, 12489 Berlin, Germany <br>
5Cellenion SASU, 60 Avenue Rockefeller, Bâtiment BioSerra2, 69008 Lyon, France <br>
6Present address: Department of Microchemistry, Proteomics, Lipidomics and Next Generation Sequencing, 
Genentech, 1 DNA Way, South San Francisco, 94080, United States <br>
*Correspondence: Dr. Ying Zhu, zhu.ying@gene.com and Dr. James M. Fulcher james.fulcher@pnnl.gov <br>
## Abstract
Single-cell multiomics provides comprehensive insights into gene regulatory networks, cellular diversity, and temporal dynamics. While tools for co-profiling single-cell genomes, transcriptomes, or epigenomes are available, accessing proteomes in parallel is more challenging. We developed nanoSPLITS (nanodroplet SPlitting for Linked-multimodal Investigations of Trace Samples), an integrated platform that enables global profiling of the transcriptome and proteome from same single cells using RNA sequencing and mass spectrometry-based proteomics, respectively. NanoSPLITS can precisely quantify over 5000 genes, 2000 proteins, and 130 phosphopeptides per single cell and identify cell-type-specific markers from these modalities. We demonstrated nanoSPLITS-based single-cell multiomics can provide comprehensive molecular characterizations of both secondary and primary cells, with insights including covarying functional protein/gene clusters and unique phosphorylation events. Finally, we introduced an efficient approach to utilize nanoSPLITS multiomic datasets for annotating unknown cell identities and detecting their protein markers by mapping transcriptomic data to existing single-cell reference databases.

![alt text](https://github.com/Cajun-data/nanoSPLITS/blob/main/Scheme1_Edited_new.png)


