---
title: "nanoSPLITS C10/SVEC Single Cell Experiment - R Markdown File (Figure 3 and associated supplementary figures)"
author: "James M. Fulcher"
date: "Last updated: 11/21/2023"
output:github_document:
  BiocStyle::html_document:
    toc_float: true
    code_folding: hide
---

# Introduction
This markdown file details the data structure and code used to generate figures
for the nanoSPLITS manuscript. For this experiment, C10 mouse type II alveolar
cells and SVEC cells were split for parallel transcriptomics and proteomics. For transcriptomics,
both raw counts and TPM normalized values were utilized. For proteomics, 
summarized protein intensities and riBAQ values were utilized. 

# Proteomics

## Data structure
We import "C10SVEC_singlecells_Protein_intensities.tsv" and "rIBAQ.RData" for the LFQ proteomic data.
"C10SVEC_singlecells_Protein_intensities.tsv" contains the summarized protein intensity values output from FragPipe (17.1). "rIBAQ.RData" contains iBAQ summarized (under variable "response") protein values which 
are converted to riBAQ values within this script. iBAQ measurements were calculated
using peptide level quant data from FragPipe outside of this script with the aLFQ R package.
 Several accessory files are needed for reference within the script ("filter.RData", "protlist.csv", and "convert_uniprot.tsv"). 

## Data Quality Control
For quality control of proteomics data, samples with less than 1,000 identified proteins were removed from downstream 
analysis. To accomplish dimensional reduction of proteomics data, values were imputed using DreamAI. For Figure 3d,  only proteins with less than 40% missingness in both SVEC and C10 were imputed and retained.  For Figure 3e, only proteins in C10 with less than 40% missingness were imputed and retained. 

# Transcriptomics

## Data structure
We import "C10SVEC_singlecells_Counts.txt" and "C10SVEC_singlecells_TPM.txt" for raw counts and
 TPM normalized expression values of the transcriptomic data, respectively. This experiment relies more heavily on raw counts, which are utilized in SCNorm in order to perform between-sample normalization. Normalized expression counts were read into the markdown file with "C10SVEC_singlecells_norm_counts.csv". 

## Data Quality Control
For quality control of transcriptomics data, separate criteria are used depending on cell type. For C10, cells with more than 3,000 genes supported by a minimum of 5 raw counts are retained. For SVEC, cells with more than 2,000 genes supported by a minimum of 5 raw counts are retained. 


# Relevant Code Chunks

## Data import
```{r Setup, include=TRUE, message = FALSE, warning = FALSE,results = 'hide'}
path <- "./"
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(rootdir = path)
setwd(path)
library(tidyverse); library(proDA);library(aLFQ);library(knitr); library(ggrepel);
library(vegan);library(corrplot); library(org.Mm.eg.db); library(AnnotationDbi);
library(corrr); library(reshape2);library(PCAtools);
library(fs);library(SC3); library(SingleCellExperiment);library(scater);
library(Seurat); library(SeuratData); library(cowplot);library(data.table);
library(viridis)

singcell <- read_tsv("C10SVEC_singlecells_Protein_intensities.tsv", show_col_types = FALSE)
RNAx <- read.delim("C10SVEC_singlecells_Counts.txt", check.names = FALSE )
RNA_norm3 <- read.csv("C10SVEC_singlecells_norm_counts.csv", check.names = FALSE)
RNAx_TPM <- read.delim("C10SVEC_singlecells_TPM.txt", check.names = FALSE)
names(RNAx_TPM)[names(RNAx_TPM) == ''] <- 'Gene'
RNAx_counts <- read.delim("C10SVEC_singlecells_Counts.txt", check.names = FALSE)
names(RNAx_counts)[names(RNAx_counts) == ''] <- 'Gene'

load("rIBAQ.RData")
load("filter.RData")

protlist <- read.csv("protlist.csv")
convert <- read_tsv("convert_uniprot.tsv")
s.genes <- read.csv("sgenes.csv", header = FALSE,
                    check.names = FALSE)$V1
g2m.genes <- read.csv("g2mgenes.csv", header = FALSE)$V1


```



```{r, message = FALSE, warning = FALSE,results = 'hide'}


singcell2 <- singcell %>%
  filter(!grepl("contam_sp", PROTID))
singcell2[singcell2 == 0] <- NA


singcell2[,2:71] <- log2(singcell2[,2:71])
singcell_long1 <- singcell2 %>%
  pivot_longer(!PROTID, names_to = "SampleID", values_to = "Intensity") %>%
  filter(!is.na(Intensity))


names(RNAx)[names(RNAx) == ''] <- 'Gene'
RNAx2 <- RNAx %>%
  pivot_longer(!Gene, names_to = "SampleID", values_to = "Counts") %>%
  filter(Counts > 4)


RNAx3 <- RNAx2 %>%
    mutate(Type = case_when(grepl("05J", SampleID) ~ "C10",
                            grepl("07J", SampleID) ~ "SVEC")) %>%
  group_by(SampleID) %>%
  add_count(name = "n") %>%
  mutate(Filter1 = case_when(Type == "C10" & n >= 3000 ~ "Keep",
                             Type == "SVEC" & n >= 2000 ~ "Keep",
                             TRUE ~ "Remove")) %>%
  filter(Filter1 == "Keep") %>%
  ungroup() %>%
  dplyr::select(-n, -Type, -Filter1)


```




## Post-normalization of scRNAseq data with SCNorm
```{r, message = FALSE, warning = FALSE,results = 'hide'}


names(RNA_norm3)[names(RNA_norm3) == ''] <- 'Gene' 

 RNA_norm3 <- RNA_norm3 %>%
   remove_rownames %>% column_to_rownames(var= "Gene" ) 


RNAx4 <- RNA_norm3 %>%
  tibble::rownames_to_column(var = "Gene") %>%
  pivot_longer(!Gene, names_to = "SampleID", values_to = "Counts") %>%
  filter(Counts >= 5)


```


## Convert UNIPROT accessions to gene symbols, and perform inference
```{r, message = FALSE, warning = FALSE,results = 'hide'}

UniAc <- unique(singcell_long1$PROTID)

geneSymbols <- select(org.Mm.eg.db, keys=UniAc, columns= c("SYMBOL","UNIPROT"),
                      keytype="UNIPROT", multiVals="first")

geneSymbols <- geneSymbols %>%
  mutate(Gene = SYMBOL) %>%
  dplyr::select(Gene,UNIPROT) 

convert <- read_tsv("convert_uniprot.tsv")

geneSymbols <- full_join(geneSymbols, convert, by = c("Gene","UNIPROT")) %>%
  filter(!is.na(Gene))

singcell_long2 <- singcell_long1 %>%
  filter(!is.na(Intensity)) %>%
  mutate(UNIPROT = PROTID)

singcell_long2 <- full_join(singcell_long2,geneSymbols, by = "UNIPROT") %>%
  mutate(Gene = case_when(is.na(Gene) ~ UNIPROT,
                          TRUE ~ Gene)) %>%
  group_by(SampleID, Gene) %>%
  slice_max(order_by = Intensity, n = 1 , with_ties = FALSE)  %>%
  ungroup()


###### Inference

combined3 <- full_join(singcell_long2, RNAx4, by = c("SampleID", "Gene")) 


inference1 <- combined3 %>%
  filter(!is.na(Intensity)) %>%
  group_by(SampleID, Intensity) %>%
  add_count(name = "n") %>%
  filter(n > 1) %>%
  ungroup() %>%
  distinct(Gene,UNIPROT)



inference2 <- combined3 %>%
  filter(!is.na(Intensity) & !is.na(Counts)) %>%
  group_by(SampleID, Intensity) %>%
  add_count(name = "n") %>%
  filter(n > 1) %>%
  distinct(n, Gene, Counts, UNIPROT) %>%
  group_by(Gene) %>%
  add_count(name  = "n2") %>%
  group_by(Gene) %>%
  slice_max(order_by = n2, n = 1) %>%
  mutate(med = median(Counts)) %>%
  group_by(UNIPROT) %>%
  slice_max(order_by = med, n = 1, with_ties = FALSE) %>%
  ungroup() %>%
  distinct(UNIPROT, Gene)

inference3 <- inference1 %>%
  filter(!Gene %in% inference2$Gene)

singcell_long3 <- singcell_long2 %>%
  filter(!Gene %in% inference3$Gene) %>%
  filter(!is.na(SampleID))
```




```{r, message = FALSE, warning = FALSE,results = 'hide'}
##Filter samples with high keratin contam
removeID <- singcell_long3 %>%
    mutate(keratin = case_when(grepl("Krt", Gene) ~ "Yes",
                             TRUE ~ "No")) %>%
    filter(keratin == "Yes") %>%
  group_by(SampleID) %>%
  summarize(med = median(Intensity)) %>%
  filter(med > 20) %>%
  distinct(SampleID)

```


```{r, message = FALSE, warning = FALSE,results = 'hide'}

singcell3 <- singcell_long3 %>%
  mutate(Type = case_when(grepl("05J", SampleID) ~ "C10",
                          grepl("07J", SampleID) ~ "SVEC")) %>%
  filter(!grepl("Krt", Gene)) %>%
  filter(!SampleID %in% removeID$SampleID) %>%
  group_by(SampleID) %>%
  add_count(name = "n") %>%
  filter(n >= 1000) %>%
  dplyr::select(-n) %>%
  pivot_wider(id_cols = "Gene", names_from = "SampleID",
              values_from = "Intensity")



C10cell <- singcell3 %>%
  dplyr::select(starts_with("Gene")| starts_with("05J")) 

C10cell <- C10cell %>% remove_rownames %>% column_to_rownames(var="Gene")
C10cell <- as.data.frame(median_normalization(as.matrix(C10cell)))
C10cell_long <- C10cell %>%
  rownames_to_column(., "Gene") %>%
  pivot_longer(!Gene, names_to = "SampleID", values_to = "Intensity")

SVECcell <- singcell3 %>%
  dplyr::select(starts_with("Gene")| starts_with("07J"))

SVECcell <- SVECcell %>% remove_rownames %>% column_to_rownames(var="Gene")

SVECcell <- as.data.frame(median_normalization(as.matrix(SVECcell)))
SVECcell_long <- SVECcell %>%
  rownames_to_column(., "Gene") %>%
  pivot_longer(!Gene, names_to = "SampleID", values_to = "Intensity")

singcell_long <- rbind(C10cell_long,SVECcell_long) %>%
  filter(!is.na(Intensity))

```




## Intermediate step to combine scRNAseq and scProteomic data
```{r, message = FALSE, warning = FALSE,results = 'hide'}


RNAx5 <- RNAx4 %>%
  mutate(SampleID = paste0(SampleID, "_RNA")) %>%
  mutate(Intensity = log2(Counts)) %>%
  filter(!is.na(Counts)) %>%
  ungroup()



combined4 <- full_join(singcell_long, RNAx5) %>%
  filter(!is.na(SampleID)) %>%
  dplyr::select(-Counts)



samp_filter2 <- unique(RNAx4$SampleID)
samp_filter <- unique(singcell_long$SampleID)


```


## Figure 3b code
```{r, message = FALSE, warning = FALSE,results = 'hide'}




RNAx2 <- RNAx_TPM %>%
  pivot_longer(!Gene, names_to = "SampleID", values_to = "TPM") 

RNAx3 <- RNAx_counts %>%
  pivot_longer(!Gene, names_to = "SampleID", values_to = "Counts") 

RNAx4 <- inner_join(RNAx2, RNAx3)


correlation_gene <- inner_join(singcell_long, RNAx4) %>%
  filter(!is.na(TPM)) %>%
  filter(TPM >= 0.1) %>%
  mutate(TPM = log2(TPM))

datatest <- correlation_gene %>%
  mutate(Type = case_when(grepl("C10_", SampleID) ~ "C10",
                          grepl("SVEC", SampleID) ~ "SVEC")) %>%
  filter(Type  == "C10" | Type == "SVEC") %>%
  group_by(Gene, Type) %>%
  add_count(name = "Obs") %>%
  filter(Obs >= 4) %>%
  ungroup()



datatest <- as_tibble(datatest)


data_nest <- group_by(datatest,Type, Gene) %>% nest()

cor_fun <- function(df) cor.test(df$Intensity, df$TPM, method = "pearson") %>%
  broom::tidy()

data_nest <- mutate(data_nest, model = map(data, cor_fun))
corr_pr <- dplyr::select(data_nest, -data) %>% unnest(cols = c(model))
corr_pr <- corr_pr %>%
  ungroup() %>%
  mutate(FDR = p.adjust(`p.value`, method = "BH")) %>%
  mutate(Type2 = "mRNA-Protein")

set.seed(524)
datatest <- correlation_gene %>%
  mutate(Type = case_when(grepl("C10_", SampleID) ~ "C10",
                          grepl("SVEC", SampleID) ~ "SVEC")) %>%
  filter(Type  == "C10" | Type == "SVEC") %>%
  group_by(Gene, Type) %>%
  add_count(name = "Obs") %>%
  filter(Obs >= 4) %>%
  transform(TPM = sample(TPM)) %>%
  transform(Intensity = sample(Intensity)) %>%
  ungroup()

datatest <- as_tibble(datatest)

data_nest <- group_by(datatest,Type, Gene) %>% nest()

cor_fun <- function(df) cor.test(df$Intensity, df$TPM, method = "pearson") %>%
  broom::tidy()

data_nest <- mutate(data_nest, model = map(data, cor_fun))
corr_pr2 <- dplyr::select(data_nest, -data) %>% unnest(cols = c(model))
corr_pr2 <- corr_pr2 %>%
  ungroup() %>%
  mutate(FDR = p.adjust(`p.value`, method = "BH")) %>%
  mutate(Type2 = "Randomized")

combined_pearson <- full_join(corr_pr, corr_pr2)

main1 <- combined_pearson %>%
  ggplot()+
  aes(x = estimate, fill = Type2)+
  geom_histogram(alpha = 0.6,
                 position = "identity",
                 bins = 100)+
  xlab("")+
  ylab("Genes (n)")+
  theme_minimal(base_size = 24)+
  theme(panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line())+
  scale_fill_brewer(name = "", palette = 2,type = "qual",
                    direction = -1)+
  theme(legend.position = "bottom",
        legend.direction = "vertical",
        legend.box = "horizontal")+
  geom_vline(xintercept = 0)


 wilcox.test(corr_pr$estimate, corr_pr2$estimate, alternative = "two.sided")

```


## Figure 3c,d code
```{r, message = FALSE, warning = FALSE,results = 'hide'}

  
C10SVEC.rna <- read.csv(file = "C10SVEC_singlecells_norm_counts.csv", sep = ",",
                          header = TRUE, row.names = 1, check.names = FALSE) %>%
  as.sparse()

names <- C10SVEC.rna@Dimnames[[2]]


C10SVEC.prot <- read.csv(file = "C10SVEC_singlecells_protein_impute.csv", sep = ",",
                         header = TRUE, row.names = 1, check.names = FALSE) %>%
  as.sparse()


names2 <- C10SVEC.prot@Dimnames[[2]]

C10SVEC_sub.rna <- read.csv(file = "C10SVEC_singlecells_norm_counts.csv", sep = ",",
                            header = TRUE, row.names = 1, check.names = FALSE) %>%
  dplyr::select(one_of(names2)) %>%
  as.sparse()

check <- colnames(C10SVEC_sub.rna)



C10SVEC_sub.prot <- read.csv(file = "C10SVEC_singlecells_protein_impute.csv", sep = ",",
                             header = TRUE, row.names = 1, check.names = FALSE) %>%
  dplyr::select(one_of(names))


C10SVEC_sub.prot <- C10SVEC_sub.prot[ , check] %>%
  as.sparse()



C10SVEC <- CreateSeuratObject(counts = C10SVEC_sub.rna)

prot_assay <- CreateAssayObject(counts = C10SVEC_sub.prot)

C10SVEC[["Prot"]] <- prot_assay

DefaultAssay(C10SVEC) <- 'RNA'

C10SVEC <- NormalizeData(C10SVEC) 

C10SVEC <- FindVariableFeatures(C10SVEC,
                                selection.method = "disp")
C10SVEC <- ScaleData(C10SVEC)
C10SVEC <- RunPCA(C10SVEC, verbose = FALSE,
                  npcs = 48)
C10SVEC <- FindNeighbors(C10SVEC, dims = 1:4)
C10SVEC <- FindClusters(C10SVEC, resolution = 1, verbose = FALSE)
C10SVEC <- RunUMAP(C10SVEC, dims = 1:4)


DefaultAssay(C10SVEC) <- 'Prot'

VariableFeatures(C10SVEC) <- rownames(C10SVEC[["Prot"]])
C10SVEC <-  ScaleData(C10SVEC) %>% RunPCA(reduction.name = 'apca',
                                          npcs = 48)

C10SVEC <- RenameIdents(C10SVEC, "0" = '  C10', "1" = 'SVEC  ')
levels(C10SVEC)
C10SVEC$celltype <- Idents(C10SVEC)
C10SVEC <- FindMultiModalNeighbors(
  C10SVEC, reduction.list = list("pca", "apca"), 
  dims.list = list(1:4, 1:3),
  knn.range = 48, smooth = FALSE,
  return.intermediate = TRUE,
  modality.weight.name = c("RNA.weight", "Prot.weight"))

C10SVEC <- RunUMAP(C10SVEC, nn.name = "weighted.nn", reduction.name = "wnn.umap", reduction.key = "wnnUMAP_")
C10SVEC <- FindClusters(C10SVEC, graph.name = "wsnn", algorithm = 1, resolution = 1, verbose = FALSE)

C10SVEC <- RenameIdents(C10SVEC, "0" = 'C10    ',"1" = "   SVEC")
levels(C10SVEC)
C10SVEC$celltype <- Idents(C10SVEC)


p1 <- DimPlot(C10SVEC, group.by = "celltype",
              reduction = 'wnn.umap', label = FALSE, repel = TRUE, label.size = 5,
              pt.size = 3.5,
              cols = c("#00D65C","#82B7FF")) + NoLegend()+
  scale_x_continuous(limits = c(-7, 7))+
  scale_y_continuous(limits = c(-8,9))

p2 <- FeaturePlot(C10SVEC,label = FALSE, features = c("rna_H2-K1"),
                  reduction = 'wnn.umap', max.cutoff = 100, 
                  cols = c("lightgrey", "purple"),
                  ncol = 1,
                  repel = TRUE,
                  order = TRUE,
                  slot = "scale.data",
                  pt.size = 3.5)+
  scale_x_continuous(limits = c(-7, 7))+
  scale_y_continuous(limits = c(-8,9))

p3 <- FeaturePlot(C10SVEC,label = FALSE, features = c("prot_H2-K1"),
                  reduction = 'wnn.umap', max.cutoff = 100, 
                  ncol = 1,
                  cols = c("lightgrey", "red"),
                  repel = TRUE,
                  order = TRUE,
                  slot = "scale.data",
                  pt.size = 3.5)+
  scale_x_continuous(limits = c(-7, 7))+
  scale_y_continuous(limits = c(-8,9))

```



## Figure 3a code
```{r, message = FALSE, warning = FALSE,results = 'hide'}

test2 <- test %>%
  group_by(run_id) %>%
  mutate(Total = sum(response)) %>%
  ungroup() %>%
  mutate(riBAQ = response/Total) %>%
  group_by(protein_id) %>%
  summarize(median = median(riBAQ))

test2$Rank <- rank(-test2$median)

test3 <- test %>%
  group_by(run_id) %>%
  mutate(Total = sum(response)) %>%
  ungroup() %>%
  mutate(riBAQ = response/Total) %>%
  mutate(SampleID = run_id) %>%
  inner_join(., protlist) %>%
  dplyr::select(SampleID, PROTID, riBAQ)



test3 <- test3 %>%
  group_by(SampleID) %>%
  add_count(name = "n") %>%
  filter(n >= 1000) %>%
  dplyr::select(-n) 



RNAx2 <- RNAx_TPM %>%
  pivot_longer(!Gene, names_to = "SampleID", values_to = "TPM") 

RNAx3 <- RNAx_counts %>%
  pivot_longer(!Gene, names_to = "SampleID", values_to = "Counts") 

RNAx4 <- inner_join(RNAx2, RNAx3)


RNAx5 <- RNAx4 %>%
  filter(Counts >= 5) %>%
  mutate(Type = case_when(grepl("05J", SampleID) ~ "C10",
                          grepl("07J", SampleID) ~ "SVEC")) %>%
  group_by(SampleID) %>%
  add_count(name = "n") %>%
  mutate(Filter1 = case_when(Type == "C10" & n >= 3000 ~ "Keep",
                             Type == "SVEC" & n >= 2000 ~ "Keep",
                             TRUE ~ "Remove")) %>%
  filter(Filter1 == "Keep") %>%
  ungroup() %>%
  dplyr::select(-Counts, -n, -Type, -Filter1)


UniAc <- unique(test3$PROTID)

geneSymbols <- select(org.Mm.eg.db, keys=UniAc, columns= c("SYMBOL","UNIPROT"),
                      keytype="UNIPROT", multiVals="first")


geneSymbols <- geneSymbols %>%
  mutate(Gene = SYMBOL) %>%
  dplyr::select(Gene,UNIPROT) 



geneSymbols <- full_join(geneSymbols, convert, by = c("Gene","UNIPROT")) %>%
  filter(!is.na(Gene))

test4 <- test3 %>%
  mutate(UNIPROT = PROTID)


x_long2 <- inner_join(test4,geneSymbols, by = "UNIPROT") %>%
  mutate(Gene = case_when(is.na(Gene) ~ UNIPROT,
                          TRUE ~ Gene)) %>%
  group_by(SampleID, Gene) %>%
  mutate(logriBAQ = log2(riBAQ)) %>%
  slice_max(order_by = logriBAQ, n = 1 , with_ties = FALSE)  %>%
  ungroup() 


###### Inference

combined3 <- full_join(x_long2, RNAx5, by = c("SampleID", "Gene")) 


inference1 <- combined3 %>%
  filter(!is.na(logriBAQ)) %>%
  group_by(SampleID, logriBAQ) %>%
  add_count(name = "n") %>%
  filter(n > 1) %>%
  ungroup() %>%
  distinct(Gene,UNIPROT)



inference2 <- combined3 %>%
  filter(!is.na(logriBAQ) & !is.na(TPM)) %>%
  group_by(SampleID, logriBAQ) %>%
  add_count(name = "n") %>%
  filter(n > 1) %>%
  distinct(n, Gene, TPM, UNIPROT) %>%
  group_by(Gene) %>%
  add_count(name  = "n2") %>%
  group_by(Gene) %>%
  slice_max(order_by = n2, n = 1) %>%
  mutate(med = median(TPM)) %>%
  group_by(UNIPROT) %>%
  slice_max(order_by = med, n = 1, with_ties = FALSE) %>%
  ungroup() %>%
  distinct(UNIPROT, Gene)

inference3 <- inference1 %>%
  filter(!Gene %in% inference2$Gene)

x_long3 <- x_long2 %>%
  filter(!Gene %in% inference3$Gene)


RNAx6 <- RNAx5 %>%
  mutate(SampleID = paste0(SampleID, "_RNA")) %>%
  mutate(logriBAQ = log2(TPM)) %>%
  filter(!is.na(TPM)) %>%
  ungroup()

combined4 <- full_join(x_long3, RNAx6) %>%
  filter(!is.na(SampleID)) %>%
  filter(grepl(paste(datafilter_ibaq, collapse = "|"), SampleID)) %>%
  filter(!SampleID %in%  c("05J_C10_B12_RNA","07J_SVEC_A11_RNA",
                           "05J_C10_D10_RNA","05J_C10_D12_RNA",
                           "07J_SVEC_B12_RNA", "07J_SVEC_B12")) %>%
  dplyr::select(-TPM, -UNIPROT, -PROTID, -riBAQ)


corx <- combined4 %>%
  spread(SampleID, logriBAQ) %>%
  ungroup() 

corx2 <- corx %>% remove_rownames %>% column_to_rownames(var="Gene")

corTest <- cor(corx2, use = "pairwise.complete.obs") 



```


# Main Figures

## Figure 3a. Pearson correlation heatmap of SVEC and C10 cells (and associated code)
```{r, message=FALSE, warning=FALSE, dpi=300, fig.width= 10, fig.height = 10, results='hide'}



corrplot(corTest, method = 'shade', order = 'hclust',
         hclust.method = 'centroid',
         is.corr = TRUE, tl.srt = 45,
         #col.lim = c(min(corTest),max(corTest)),
         diag = FALSE, 
         #type = "lower",
         tl.col = "black",
         cl.pos = "b",
         cl.cex = 2,
         col.lim = c(0, 1),
         tl.pos = "n"
) %>%
  corrRect(c(0, 27, 50,
             73,99), col = 'white',
           lwd = 10)+
  theme(text=element_text(family="Helvetica"))


```